libs/libapi/src/libapi/duckdb.py

```diff
@@ -67,25 +72,39 @@ def get_download_folder(root_directory: StrPath, dataset: str, revision: str, co
     return f"{root_directory}/{DUCKDB_INDEX_DOWNLOADS_SUBDIRECTORY}/{DUCKDB_VERSION}/{subdirectory}"
 
 
+def available_disk_space(save_path: str, required_space: int) -> bool:
+    disk_stat = os.statvfs(save_path)
+    # Calculate free space in bytes
+    free_space = disk_stat.f_bavail * disk_stat.f_frsize
+    return free_space >= required_space
+
+
 def download_index_file(
     cache_folder: str,
     index_folder: str,
     target_revision: str,
+    size_bytes: int,
     dataset: str,
     repo_file_location: str,
     hf_token: Optional[str] = None,
 ) -> None:
     logging.info(f"init_dir {index_folder}")
-    init_dir(index_folder)
-    download_file_from_hub(
-        repo_type=REPO_TYPE,
-        revision=target_revision,
-        repo_id=dataset,
-        filename=repo_file_location,
-        local_dir=index_folder,
-        hf_token=hf_token,
-        cache_dir=cache_folder,
-    )
+    if not available_disk_space(cache_folder, size_bytes):
+        raise DownloadIndexError("Unable to download index file")  # Should we show the remaining disk space?
```

Comment:
I was unsure what exactly we would like to show the user. Getting a lack of storage should be something to alert in our internal process.

Reply:
It returns a `HTTPStatus.INTERNAL_SERVER_ERROR` error (500), which is OK. We also have the `DownloadIndexError` error code, so, if a user is reporting this, we will know what occurs

Reply:
But indeed, we could improve the text a bit, to make it clearer for the user (here and in the other comment). As it refers to the internal, maybe catch this specific exception in the /search service (and in the other ones in a similar way) and return another exception with: `Cannot perform the search due to a lack of disk space on the server. Please report the issue.`

Reply:
And I imagine there is a way to see with the infra to get alerts before this occurs

Reply:
Done; I changed it to `Cannot perform the operation due to a lack of disk space on the server. Please report the issue.` because it it used for filter and search.